src/api_server/src/lib.rs

```diff
@@ -250,11 +237,10 @@ impl ApiServer {
             Ok((req_action, mut parsing_info)) => {
                 let mut response = match req_action {
                     RequestAction::Sync(vmm_action) => {
-                        self.serve_vmm_action_request(vmm_action, request_processing_start_us)
+                        let res =
+                            self.serve_vmm_action_request(vmm_action, request_processing_start_us);
+                        return res;
```

Comment:
Isn't this functionally the same but a bit uglier 😆 ?

Reply:
Yes, didn't realize from the code snippet that the function goes on for longer than the `match`.